packages/react-charting/src/components/AreaChart/AreaChart.base.tsx

```diff
@@ -368,25 +372,28 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
         allChartPoints.push(...singleChartPoint.data);
       });
 
-    let tempArr = allChartPoints;
-    while (tempArr.length) {
-      const valToCheck = tempArr[0].x instanceof Date ? tempArr[0].x.toLocaleString() : tempArr[0].x;
-      const filteredChartPoints: ILineChartDataPoint[] = tempArr.filter(
-        (point: ILineChartDataPoint) => (point.x instanceof Date ? point.x.toLocaleString() : point.x) === valToCheck,
-      );
+    const mapOfXvalToListOfDataPoints: IMapXToDataSet = {};
+    allChartPoints.forEach((dataPoint: ILineChartDataPoint) => {
+      const xValue = dataPoint.x instanceof Date ? dataPoint.x.toLocaleString() : dataPoint.x;
+      // map of x value to the list of data points which share the same x value .
+      if (mapOfXvalToListOfDataPoints[xValue]) {
+        mapOfXvalToListOfDataPoints[xValue].push(dataPoint);
+      } else {
+        mapOfXvalToListOfDataPoints[xValue] = [dataPoint];
+      }
+    });
+
+    Object.keys(mapOfXvalToListOfDataPoints).forEach((key: number | string) => {
```

Comment:
in which scenario is key possible as string

Reply:
When the x value will of string type.

Reply:
I dont think area chart can have x axis as string. It can be either number or date.

Reply:
If the xValue is of date type then we are converting it to string otherwise keeping it number.